File: pset6/sentimental/caesar.py
import sys
from cs50 import get_string

def is_int(string):
    try:
        int(string)
        return True
    except ValueError:
        return False

def main():

    #Make sure there's one system argument
    while True:
        if len(sys.argv) != 2:
            sys.exit('1')
        elif is_int(sys.argv[1]) == False:
            sys.exit('1')
        elif int(sys.argv[1]) < 0:
            sys.exit('1')
        else:
            break

    n = sys.argv[1]
    #n %= 26

    text = get_string('plaintext: ')
    ctext = ''

    for i in range(0, len(text)):
        if text[i].isalpha() is True:
            if text[i].isupper() is True:
                aindex = ord(text[i]) - 65
                cindex = (aindex + int(n)) % 26
                ctext += chr(cindex + 65)
            elif text[i].islower() is True:
                aindex = ord(text[i]) - 97
                cindex = (aindex + int(n)) % 26
                ctext += chr(cindex + 97)
            else:
                ctext += text[i]

        else:
            ctext += text[i]

    print ('ciphertext: ' + ctext)



if __name__ == "__main__":
    main()

File: pset2/caesar.c
//Caesar -

#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
//#include <stdlib.h>

//Take in a command line argument
int main(int argc, string argv[])
{
    //Check the input arguments to make sure there's only one
    if (argc != 2){
        exit(1);
    }

    //Convert the input argument from a string to an integer.
    int n = atoi(argv[1]);

    //Get plaintext and start cypher
    string plaintext = get_string("plaintext: ");

    printf("ciphertext: ");

    //Loop through and print the cipher text
    //Print each character individually
    for (int i = 0, s = strlen(plaintext); i < s ;i++)
      {
          if (isalpha(plaintext[i]))
            {
                if(isupper(plaintext[i]))
                    {
                        int alphabetindex = plaintext[i] - 65;
                        int ciphertext = (alphabetindex + n) % 26;
                        char ciphertextASCII = ciphertext + 65;
                        printf("%c",ciphertextASCII);
                    }
                if(islower(plaintext[i]))
                    {
                        int alphabetindex = plaintext[i] - 97;
                        int ciphertext = (alphabetindex + n) % 26;
                        char ciphertextASCII = ciphertext + 97;
                        printf("%c",ciphertextASCII);
                    }
            }
            else
            {
                printf("%c",plaintext[i]);
            }

          //char ciphertext = plaintext[i] + n;
          //printf("%c",ciphertext);

          //Debugg:  printf("%c\n", plaintext[i]);
      }
     printf("\n");
     return 0;
}